src/components/templates/loginForm/loginForm.tsx

```diff
@@ -1,70 +1,98 @@
-import React from 'react';
+import React, {useState, useEffect} from 'react';
 import InputText from '../../atoms/inputs'
 import ButtonComponent from '../../atoms/button'
-import {Form} from 'semantic-ui-react'
-import { action } from '@storybook/addon-actions';
-import { StyledFormField } from './style/loginForm.styled.component';
+// import {Form} from 'semantic-ui-react'
+import {
+    Form,
+    SubmitButton,
+    Input
+  } from "formik-semantic-ui-react";
+import { Formik } from "formik";
+import * as Yup from "yup";
 
 
-export type loginFormInterface = {
+export type IloginForm = {
     isDisabled:boolean,
     isPrimary:boolean,
-    isLoading:boolean,
-    isError?:boolean,
+    isDisabledLogBtn?:boolean
 }
 
 const LogInForm = (
     {
         isDisabled = false,
         isPrimary = true,
-        isLoading = false,
-        isError = false,
-    }:loginFormInterface) => {
+    }) => {
+    
+    useEffect(() => {
+        setDisabledState(isDisabled);
+        setIsPrimaryState(isPrimary)
+    }, [isDisabled,isPrimary]);
+
+    const [isDisabledState, setDisabledState] = useState(isDisabled);
+    const [isPrimaryState, setIsPrimaryState] = useState(isPrimary);
+
+    const initialValues = {
+        email: "",
+        password: ""
+    };
+
+    const validationSchema = Yup.object({
+        email: Yup.string().email("Invalid email address").required("Email is required"),
+        password: Yup.string()
+            .min(8, "Must be at least 8 characters")
+            .required("Password is required")
+    });
+
+
     return(
         <>
-         <Form onSubmit={action('onSubmit')}>
-            <Form.Group widths="equal">
-                <StyledFormField
-                control={InputText}
-                fluid
-                label="Email"
-                pattern="[A-Za-z0-9._%+-]{2,}@[a-zA-Z]{1,}([.]{1}[a-zA-Z]{2,}|[.]{1}[a-zA-Z]{2,}[.]{1}[a-zA-Z]{2,})"
-                placeholder="Email"
-                type="email"
-                isDisabled={isDisabled}
-                disabled={isDisabled}
-                isError={isError}
-                error ={isError}
-                isErrorProp={isError}
-                name="email"
+        <Formik
+        initialValues={initialValues}
+        validationSchema={validationSchema}
+        onSubmit={(_, { setSubmitting, resetForm }) => {
+            setDisabledState(true);
+          setTimeout(() => {
+            setSubmitting(false);
+            setDisabledState(false);
+            resetForm();
+          }, 1000);
```

Comment:
Correct me if I'm wrong but these lines of code are only to mock an api request, correct?